src/server/acl/acl_family.cc

```diff
@@ -222,6 +271,145 @@ void AclFamily::WhoAmI(CmdArgList args, ConnectionContext* cntx) {
+  auto acl_file_path = absl::GetFlag(FLAGS_aclfile);
+  if (acl_file_path.empty()) {
+    cntx->SendError("Dragonfly is not configured to use an ACL file.");
+    return;
+  }
+
+  auto res = io::OpenWrite(acl_file_path);
+  if (!res) {
+    std::string error = absl::StrCat("Failed to open the aclfile: ", res.error().message());
+    LOG(ERROR) << error;
+    cntx->SendError(error);
+    return;
+  }
+
+  std::unique_ptr<io::WriteFile> file(res.value());
+  std::string output = RegistryToString();
+  auto ec = file->Write(output);
+
+  if (ec) {
+    std::string error = absl::StrCat("Failed to write to the aclfile: ", ec.message());
+    LOG(ERROR) << error;
+    cntx->SendError(error);
+    return;
+  }
+
+  ec = file->Close();
+  if (ec) {
+    std::string error = absl::StrCat("Failed to close the aclfile ", ec.message());
+    LOG(WARNING) << error;
+    cntx->SendError(error);
+    return;
+  }
+
+  cntx->SendOk();
+}
+
+std::optional<facade::ErrorReply> AclFamily::LoadToRegistryFromFile(std::string_view full_path,
+                                                                    bool init) {
+  auto is_file_read = io::ReadFileToString(full_path);
+  if (!is_file_read) {
+    auto error = absl::StrCat("Dragonfly could not load ACL file ", full_path, " with error ",
+                              is_file_read.error().message());
+
+    LOG(WARNING) << error;
+    return {ErrorReply(std::move(error))};
+  }
+
+  auto file_contents = std::move(is_file_read.value());
+
+  if (file_contents.empty()) {
+    return {};
+  }
+
+  std::vector<std::string> usernames;
+  auto materialized = MaterializeFileContents(&usernames, file_contents);
+
+  if (!materialized) {
+    std::string error = "Error materializing acl file";
+    LOG(WARNING) << error;
+    return {ErrorReply(std::move(error))};
+  }
+
+  std::vector<User::UpdateRequest> requests;
+
+  for (auto& cmds : *materialized) {
+    auto req = ParseAclSetUser<std::vector<std::string_view>&>(cmds, true);
+    if (std::holds_alternative<ErrorReply>(req)) {
+      auto error = std::move(std::get<ErrorReply>(req));
+      LOG(WARNING) << "Error while parsing aclfile: " << error.ToSv();
+      return {std::move(error)};
+    }
+    requests.push_back(std::move(std::get<User::UpdateRequest>(req)));
+  }
+
+  auto registry_with_wlock = registry_->GetRegistryWithWriteLock();
+  auto& registry = registry_with_wlock.registry;
+  // TODO(see what redis is doing here)
+  if (!init) {
+    registry.clear();
+  }
+  std::vector<uint32_t> categories;
+  for (size_t i = 0; i < usernames.size(); ++i) {
+    auto& user = registry[usernames[i]];
+    user.Update(std::move(requests[i]));
+    categories.push_back(user.AclCategory());
+  }
+
+  if (!init) {
+    StreamUpdatesToAllProactorConnections(usernames, categories);
+  }
+
+  return {};
+}
+
+void AclFamily::Load() {
+  auto acl_file = absl::GetFlag(FLAGS_aclfile);
+  LoadToRegistryFromFile(acl_file, true);
+}
```

Comment:
So what if it fails? The instance will wrongfully start without acl

Reply:
Nope, it's fine, this is what Redis is doing. Otherwise we have to crash, I don't want to crash DF for that. 

Moreover, you can set the ACL file dynamically, via CONFIG SET

Reply:
We issue a warning and that's enough information :)